example/frontend-testing/rest.js

```diff
@@ -0,0 +1,197 @@
+    } else {
+        const responseContainer = document.getElementById('checkHealth');
+        responseContainer.textContent = "Please enter a valid API key first.";
+    }
+}
+
+function allowDrop(event) {
+    event.preventDefault();  // Allow file to be dropped
+}
+
+function handleDrop(event) {
+    event.preventDefault();
+    const files = event.dataTransfer.files;
+    if (files.length > 0) {
+        selectedFile = files[0];
+        document.getElementById('dropZone').textContent = `Selected file: ${selectedFile.name}`;
+    }
+}
+
+function handleFileSelect(event) {
+    const files = event.target.files;
+    if (files.length > 0) {
+        selectedFile = files[0];
+        document.getElementById('dropZone').textContent = `Selected file: ${selectedFile.name}`;
+    }
+}
+
+document.getElementById('dropZone').addEventListener('click', () => {
+    document.getElementById('fileInput').click();  // Open file selector dialog
+});
+
+/**
+ * Function to start transcription by sending a selected file to the API.
+ * Displays the response or error in the 'response' element.
+ */
+async function requestTranscription() {
+    const responseContainer = document.getElementById('response');
+
+    // Check if a file is selected
+    if (!selectedFile) {
+        responseContainer.textContent = 'Please upload a file first.';
+        return;
+    }
+
+    const formData = new FormData();
+
+    const apiKey = localStorage.getItem('apiKey');  // Retrieve the saved API key
+    if (!apiKey) {
+        responseContainer.textContent = 'API key is missing. Please save your API key first.';
+        return;
+    }
+
+    // Verify selected file is valid
+    if (selectedFile === null || selectedFile === undefined) {
+        responseContainer.textContent = 'No file selected.';
+        return;
+    }
+    
+    // Append necessary data to FormData object
+    formData.append('file', selectedFile);
+    formData.append('test', 'test');
+    formData.append('language', 'en');
+    formData.append('settings', '{"test": "test"}');
+    formData.append('model', 'large-v3');
+
+    responseContainer.textContent = 'Calling API...';
+
+    try {
+        // Make POST request to transcription API
+        const response = await fetch('http://localhost:8393/transcriptions', {
+            method: 'POST',
+            headers: {
+                'Authorization': `${apiKey}`
+            },
+            body: formData
+        });
+
+        // Check response status
+        if (!response.ok) {
+            throw new Error(`Error: ${response.status}`);
+        }
+
+        // Parse and display response data
+        const data = await response.json();
+        responseContainer.textContent = JSON.stringify(data, null, 2);
+        transcriptionFileName.textContent = "File name: " + selectedFile.name;
+        responseTextContainer.textContent = data['transcription_id'];
+    } catch (error) {
```

Comment:
same as above. Good idea, I dont know if it contributes to code clarity in this case